central/scannerdefinitions/handler/handler.go

```diff
@@ -381,31 +394,41 @@ func (h *httpHandler) openMostRecentDefinitions(ctx context.Context, uuid string
 	return
 }
 
-func (h *httpHandler) openMostRecentMappings(fileName string) (file *vulDefFile, err error) {
+func (h *httpHandler) openMostRecentFile(updaterKey string, fileName string) (file *vulDefFile, err error) {
 	// TODO(ROX-20520): enable fetching offline file
-	log.Info("Getting v4 repository mapping data updater")
-	u := h.getUpdater(mappingUpdaterKey)
-
+	log.Infof("Getting v4 data for updater key: %s", updaterKey)
+	u := h.getUpdater(updaterKey)
 	var onlineFile *vulDefFile
-	onlineZipFile, onlineTime, err := h.startUpdaterAndOpenFile(u)
-	if err != nil || onlineZipFile == nil {
-		return
+	openedFile, onlineTime, err := h.startUpdaterAndOpenFile(u)
+	if err != nil {
+		return nil, err
+	}
+	if openedFile == nil {
+		return nil, errors.New("zip file is nil")
```

Comment:
```suggestion
	// Ensure the updater is running.
	u.Start()
	openedFile, onlineTime, err := u.file.Open()
	if err != nil {
		return nil, err
	}
	if openedFile == nil {
		return nil, errors.New("zip file is nil")
```

A delete `startUpdaterAndOpenFile()`.